docs/hack_loki_operator.md

```diff
@@ -0,0 +1,222 @@
+# Hacking on Loki Operator
+
+Loki Operator is the Kubernetes Operator for [Loki](https://grafana.com/docs/loki/latest/) provided by the Red Hat OpenShift engineering team.
+
+## Hacking on Loki Operator using kind
+
+[kind](https://kind.sigs.k8s.io/docs/user/quick-start/) is a tool for running local Kubernetes clusters using Docker container "nodes". kind was primarily designed for testing Kubernetes itself, but may be used for local development or CI.
+
+### Requirements
+
+* Install [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) or [Openshift CLI](https://docs.openshift.com/container-platform/latest/cli_reference/openshift_cli/getting-started-cli.html) for communicating with the cluster. The guide below will be using `kubectl` for the same.
+* Create a running Kubernetes cluster using kind.
+* A container registry that you and your Kubernetes cluster can reach. We recommend  [quay.io](https://quay.io/signin/).
+
+### Installation of Loki Operator
+
+* Build and push the container image and then deploy the operator with:
```

Comment:
@sasagarw to me, hacking includes how to run the Loki operator from my laptop "local process" and not push the images to a remote location and load into Kind or OCP. 

What I do in order to develop is similar to shat you explain here but allows me a much quicker process because I do not need to push and pull images all the time ... also, this allows me to use a debugger to fix problems and progress with the code. Can you explain how to do that with Kind ... this is much more realistic than using `make deploy`. I have to say that the final stage after I am happy with the code is exactly what you explain here, I push and test ... but this is something that I typically do just very close to code ready.

Reply:
Are you referring to something along the lines of a `make run` option? If so, wouldn't it just be better to have a PR which adds the option and then add the command to the docs here versus adding all the instruction here?

Reply:
@eranra For now we will support only `kind` and `openshift`. The full cycle with using images is that you ensure that what you build runs also in containers. A personal registry is maybe the tax to pay, but nowadays everybody has simple access to one.

Reply:
Hi @periklis I do not have any problem with supporting only kind and OCP. Totally agree with that and actually, this should be good enough for most cases, so this is good ... my point is that the development process for operators requires changes to the code and checking that it works ... iteratively, and for that to be efficient, running the operator as a local process is something we are doing a lot. (i.e. `make run` ). I suggest extending the hacking explanation to talk on `make run` and how to use that process. To me, this is the most common way to "hack" the Loki operator, and very important to explain how to do that.